pipeline/api/cloud.py

```diff
@@ -434,6 +435,17 @@ def upload_pipeline(
         """
         if new_pipeline_graph._has_run_startup:
             raise Exception("Cannot upload a pipeline that has already been run.")
+        if sys.version_info.major != 3 or (
+            sys.version_info.major == 3 and sys.version_info.minor != 9
+        ):
+            print(
```

Comment:
If we're certain that non-3.9 pipelines will never run we should stop execution here, e.g. raise an exception.

Reply:
In the past some 3.8 pipelines have worked for me which is why I chose `print` vs `raise`. Ideally actually we'd be using the `logger` stuff so it could be a proper debugging warning.

Reply:
I see. I'd still be tempted to raise as running this with non-3.9 is effectively undefined behaviour, but leave that up to you.